diff --git bfd/config.bfd bfd/config.bfd
index e7b55b0dd78..4614e5b81be 100644
--- bfd/config.bfd
+++ bfd/config.bfd
@@ -231,6 +231,11 @@ case "${targ}" in
     targ_archs="$targ_archs bfd_i386_arch bfd_powerpc_arch bfd_rs6000_arch"
     want64=true
     ;;
+  aarch64*-*-bolthur*)
+    targ_defvec=aarch64_elf64_le_vec
+    targ_selvecs="aarch64_elf64_be_vec aarch64_elf32_le_vec aarch64_elf32_be_vec arm_elf32_le_vec arm_elf32_be_vec"
+    want64=true
+    ;;
   aarch64-*-elf | aarch64-*-rtems*)
     targ_defvec=aarch64_elf64_le_vec
     targ_selvecs="aarch64_elf64_be_vec aarch64_elf32_le_vec aarch64_elf32_be_vec arm_elf32_le_vec arm_elf32_be_vec"
@@ -384,6 +389,10 @@ case "${targ}" in
     targ_defvec=arm_elf32_le_vec
     targ_selvecs=arm_elf32_be_vec
     ;;
+  arm*-*-bolthur*)
+    targ_defvec=arm_elf32_le_vec
+    targ_selvecs="arm_elf32_fdpic_le_vec arm_elf32_be_vec arm_elf32_fdpic_be_vec"
+    ;;
   arm-*-elf | arm*-*-freebsd* | arm*-*-linux-* | arm*-*-conix* | \
   arm*-*-uclinux* | arm-*-kfreebsd*-gnu | \
   arm*-*-eabi* | arm-*-rtems* | arm*-*-uclinuxfdpiceabi)
diff --git config.sub config.sub
index b0f8492348d..aaed7990a42 100755
--- config.sub
+++ config.sub
@@ -1722,7 +1722,7 @@ case $os in
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
-	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx*)
+	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | bolthur*)
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
diff --git gas/configure.tgt gas/configure.tgt
index 6f46e0a6329..a3b9f726198 100644
--- gas/configure.tgt
+++ gas/configure.tgt
@@ -6,12 +6,12 @@
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 3 of the License, or
 # (at your option) any later version.
-# 
+#
 # This program is distributed in the hope that it will be useful,
 # but WITHOUT ANY WARRANTY; without even the implied warranty of
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 # GNU General Public License for more details.
-# 
+#
 # You should have received a copy of the GNU General Public License
 # along with this program; see the file COPYING3.  If not see
 # <http://www.gnu.org/licenses/>.
@@ -125,6 +125,7 @@ generic_target=${cpu_type}-$vendor-$os
 case ${generic_target} in
   aarch64*-*-elf*)			fmt=elf;;
   aarch64*-*-fuchsia*)			fmt=elf;;
+  aarch64*-*-bolthur*)			fmt=elf ;;
   aarch64*-*-linux*)			fmt=elf em=linux
     case ${cpu}-${os} in
       aarch64*-linux-gnu_ilp32)		arch=aarch64:32 ;;
@@ -161,6 +162,7 @@ case ${generic_target} in
 					fmt=coff em=wince-pe ;;
   arm-*-pe)				fmt=coff em=pe ;;
   arm-*-fuchsia*)			fmt=elf ;;
+  arm*-*-bolthur*)			fmt=elf ;;
 
   avr-*-*)				fmt=elf bfd_gas=yes ;;
 
@@ -325,7 +327,7 @@ case ${generic_target} in
   moxie-*-uclinux)			fmt=elf em=linux ;;
   moxie-*-moxiebox*)                    fmt=elf endian=little ;;
   moxie-*-*)				fmt=elf ;;
-    
+
   mt-*-elf)				fmt=elf bfd_gas=yes ;;
 
   msp430-*-*)				fmt=elf ;;
@@ -410,7 +412,7 @@ case ${generic_target} in
   wasm32-*-*)                           fmt=elf ;;
 
   xstormy16-*-*)			fmt=elf ;;
-  
+
   xgate-*-*)    			fmt=elf ;;
 
   xtensa*-*-*)				fmt=elf ;;
diff --git ld/Makefile.am ld/Makefile.am
index 2c9a487335c..695b8cb895e 100644
--- ld/Makefile.am
+++ ld/Makefile.am
@@ -165,6 +165,7 @@ ALL_EMULATION_SOURCES = \
 	earclinux_nps.c \
 	earm_wince_pe.c \
 	earmelf.c \
+	earmelf_bolthur.c \
 	earmelf_fbsd.c \
 	earmelf_fuchsia.c \
 	earmelf_linux.c \
@@ -656,6 +657,7 @@ $(ALL_EMULATION_SOURCES) $(ALL_64_EMULATION_SOURCES): $(GEN_DEPENDS)
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earclinux_nps.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earm_wince_pe.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf.Pc@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_bolthur.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_fbsd.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_fuchsia.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_linux.Pc@am__quote@
diff --git ld/configure.tgt ld/configure.tgt
index 0c780b24d57..61a6cd436d7 100644
--- ld/configure.tgt
+++ ld/configure.tgt
@@ -59,6 +59,10 @@ aarch64-*-freebsd*)	targ_emul=aarch64fbsd
 aarch64-*-fuchsia*)	targ_emul=aarch64elf
 			targ_extra_emuls="aarch64elfb armelf armelfb"
 			;;
+aarch64*-*-bolthur*)
+			targ_emul=aarch64elf
+			targ_extra_emuls="aarch64elfb armelf armelfb"
+			;;
 aarch64_be-*-linux-gnu_ilp32)
 			targ_emul=aarch64linux32b
 			targ_extra_libpath="aarch64linuxb aarch64linux aarch64linux32 armelfb_linux_eabi armelf_linux_eabi"
@@ -200,6 +204,10 @@ arm*-*-conix*)		targ_emul=armelf
 arm*-*-fuchsia*)	targ_emul=armelf_fuchsia
 			targ_extra_emuls="armelfb_fuchsia armelf armelfb"
 			;;
+arm*-*-bolthur*)
+			targ_emul=armelf_bolthur
+			targ_extra_emuls="armelfb_bolthur armelf armelfb"
+			;;
 avr-*-*)		targ_emul=avr2
 			targ_extra_emuls="avr1 avr25 avr3 avr31 avr35 avr4 avr5 avr51 avr6 avrxmega1 avrxmega2 avrxmega3 avrxmega4 avrxmega5 avrxmega6 avrxmega7 avrtiny"
 			;;
diff --git ld/emulparams/aarch64bolthur.sh ld/emulparams/aarch64bolthur.sh
new file mode 100644
index 00000000000..d035186bac0
--- /dev/null
+++ ld/emulparams/aarch64bolthur.sh
@@ -0,0 +1,4 @@
+source_sh ${srcdir}/emulparams/aarch64elf.sh
+
+GENERATE_SHLIB_SCRIPT=yes
+GENERATE_PIE_SCRIPT=yes
diff --git ld/emulparams/aarch64fbolthurb.sh ld/emulparams/aarch64fbolthurb.sh
new file mode 100644
index 00000000000..545b5ec1c9a
--- /dev/null
+++ ld/emulparams/aarch64fbolthurb.sh
@@ -0,0 +1,2 @@
+source_sh ${srcdir}/emulparams/aarch64bolthur.sh
+OUTPUT_FORMAT="elf64-bigaarch64"
diff --git ld/emulparams/armelf_bolthur.sh ld/emulparams/armelf_bolthur.sh
new file mode 100644
index 00000000000..694377d8d23
--- /dev/null
+++ ld/emulparams/armelf_bolthur.sh
@@ -0,0 +1,4 @@
+source_sh ${srcdir}/emulparams/armelf.sh
+
+GENERATE_SHLIB_SCRIPT=yes
+GENERATE_PIE_SCRIPT=yes
diff --git ld/emulparams/armelfb_bolthur.sh ld/emulparams/armelfb_bolthur.sh
new file mode 100644
index 00000000000..50b913d0638
--- /dev/null
+++ ld/emulparams/armelfb_bolthur.sh
@@ -0,0 +1,2 @@
+source_sh ${srcdir}/emulparams/armelf_bolthur.sh
+OUTPUT_FORMAT="$BIG_OUTPUT_FORMAT"
diff --git libtool.m4 libtool.m4
index e993b621ac2..7c306521825 100644
--- libtool.m4
+++ libtool.m4
@@ -1137,7 +1137,7 @@ fi
 # Invoke $ECHO with all args, space-separated.
 func_echo_all ()
 {
-    $ECHO "$*" 
+    $ECHO "$*"
 }
 
 case "$ECHO" in
@@ -2502,6 +2502,17 @@ netbsd*)
   hardcode_into_libs=yes
   ;;
 
+bolthur*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  shlibpath_var=LD_LIBRARY_PATH
+  shlibpath_overrides_runpath=no
+  hardcode_into_libs=yes
+  ;;
+
 newsos6)
   version_type=linux
   library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major $libname${shared_ext}'
@@ -3045,6 +3056,10 @@ haiku*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+bolthur*)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 hpux10.20* | hpux11*)
   lt_cv_file_magic_cmd=/usr/bin/file
   case $host_cpu in
