diff --git config.sub config.sub
index 75bb6a313..c7dd7440e 100755
--- config.sub
+++ config.sub
@@ -1363,7 +1363,8 @@ case $os in
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
-	     | midnightbsd* | amdhsa* | unleashed* | emscripten*)
+	     | midnightbsd* | amdhsa* | unleashed* | emscripten* \
+	     | bolthur*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	qnx*)
diff --git libgloss/arm/Makefile.in libgloss/arm/Makefile.in
index e1bad1710..00027a85f 100644
--- libgloss/arm/Makefile.in
+++ libgloss/arm/Makefile.in
@@ -71,7 +71,13 @@ LINUX_OBJS    = linux-syscalls0.o linux-syscalls1.o
 LINUX_SCRIPTS = linux.specs
 LINUX_INSTALL = install-linux
 
-REDBOOT_CRT0	= redboot-crt0.o 
+BOLTHUR_CRT0    = bolthur-crt0.o
+BOLTHUR_BSP     = libgloss-bolthur.a
+BOLTHUR_OBJS    = bolthur-syscalls0.o bolthur-syscalls1.o
+BOLTHUR_SCRIPTS = bolthur.specs
+BOLTHUR_INSTALL = install-bolthur
+
+REDBOOT_CRT0	= redboot-crt0.o
 REDBOOT_OBJS	= redboot-syscalls.o
 REDBOOT_SCRIPTS	= redboot.ld redboot.specs
 REDBOOT_INSTALL	= install-redboot
@@ -109,7 +115,7 @@ INCLUDES += `if [ -d ${objroot}/newlib ]; then echo -I$(srcroot)/newlib/libc/mac
 # build a test program for each target board. Just trying to get
 # it to link is a good test, so we ignore all the errors for now.
 #
-all: ${CRT0} ${LINUX_CRT0} ${LINUX_BSP} ${REDBOOT_CRT0} ${REDBOOT_OBJS} ${RDPMON_CRT0} ${RDPMON_BSP} ${RDIMON_CRT0} ${RDIMON_BSP}
+all: ${CRT0} ${LINUX_CRT0} ${LINUX_BSP} ${BOLTHUR_CRT0} ${BOLTHUR_BSP} ${REDBOOT_CRT0} ${REDBOOT_OBJS} ${RDPMON_CRT0} ${RDPMON_BSP} ${RDIMON_CRT0} ${RDIMON_BSP}
 	@rootpre=`pwd`/; export rootpre; \
 	srcrootpre=`cd $(srcdir); pwd`/; export srcrootpre; \
 	for dir in .. ${SUBDIRS}; do \
@@ -164,6 +170,10 @@ $(LINUX_BSP): $(LINUX_OBJS)
 	${AR} ${ARFLAGS} $@ $^
 	${RANLIB} $@
 
+$(BOLTHUR_BSP): $(BOLTHUR_OBJS)
+	${AR} ${ARFLAGS} $@ $^
+	${RANLIB} $@
+
 $(RDPMON_BSP): $(RDPMON_OBJS)
 	${AR} ${ARFLAGS} $@ $^
 	${RANLIB} $@
@@ -206,7 +216,7 @@ distclean maintainer-clean realclean: clean
 	rm -f Makefile config.status *~
 
 .PHONY: install info install-info clean-info
-install: ${CRT0_INSTALL} ${LINUX_INSTALL} ${REDBOOT_INSTALL} ${RDPMON_INSTALL} ${RDIMON_INSTALL} ${IQ80310_INSTALL}  ${PID_INSTALL} ${NANO_INSTALL}
+install: ${CRT0_INSTALL} ${LINUX_INSTALL} ${BOLTHUR_INSTALL} ${REDBOOT_INSTALL} ${RDPMON_INSTALL} ${RDIMON_INSTALL} ${IQ80310_INSTALL}  ${PID_INSTALL} ${NANO_INSTALL}
 	@rootpre=`pwd`/; export rootpre; \
 	srcrootpre=`cd $(srcdir); pwd`/; export srcrootpre; \
 	for dir in .. ${SUBDIRS}; do \
@@ -228,6 +238,10 @@ install-linux:
 	set -e; for x in ${LINUX_CRT0} ${LINUX_BSP}; do ${INSTALL_DATA} $$x $(DESTDIR)${tooldir}/lib${MULTISUBDIR}/$$x; done
 	set -e; for x in ${LINUX_SCRIPTS}; do ${INSTALL_DATA} ${srcdir}/${objtype}$$x $(DESTDIR)${tooldir}/lib${MULTISUBDIR}/$$x; done
 
+install-bolthur:
+	set -e; for x in ${BOLTHUR_CRT0} ${BOLTHUR_BSP}; do ${INSTALL_DATA} $$x $(DESTDIR)${tooldir}/lib${MULTISUBDIR}/$$x; done
+	set -e; for x in ${BOLTHUR_SCRIPTS}; do ${INSTALL_DATA} ${srcdir}/${objtype}$$x $(DESTDIR)${tooldir}/lib${MULTISUBDIR}/$$x; done
+
 install-redboot:
 	set -e; for x in ${REDBOOT_CRT0} ${REDBOOT_OBJS}; do ${INSTALL_DATA} $$x $(DESTDIR)${tooldir}/lib${MULTISUBDIR}/$$x; done
 	set -e; for x in ${REDBOOT_SCRIPTS}; do ${INSTALL_DATA} ${srcdir}/${objtype}$$x $(DESTDIR)${tooldir}/lib${MULTISUBDIR}/$$x; done
@@ -259,4 +273,4 @@ Makefile: Makefile.in config.status @host_makefile_frag_path@
 config.status: configure
 	$(SHELL) config.status --recheck
 
-include $(srcdir)/../multi-build.in
\ No newline at end of file
+include $(srcdir)/../multi-build.in
diff --git libgloss/arm/bolthur-crt0.c libgloss/arm/bolthur-crt0.c
new file mode 100644
index 000000000..73d3dfb1a
--- /dev/null
+++ libgloss/arm/bolthur-crt0.c
@@ -0,0 +1,88 @@
+
+/**
+ * Bolthur system startup for the ARM processor.
+ * Written by Christian Freitag <cfreitag.cf@googlemail.com>
+ * Copyright 2020 (c) Christian Freitag
+ *
+ * Permission to use, copy, modify, and distribute this software
+ * is freely granted, provided that this notice is preserved.
+ */
+
+/**
+ * Bolthur startup code for arm processor
+ */
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdnoreturn.h>
+#include "arm.h"
+
+/* forward declaration */
+int main( int argc, char* argv[], char* env[] );
+extern char** environ;
+
+/* bss start and end for cleanup */
+extern unsigned int __bss_start__;
+extern unsigned int __bss_end__;
+
+/* provided by gcc */
+#if defined( __USES_INITFINI__ )
+  extern void _init( void );
+  extern void _fini( void );
+#endif
+
+/**
+ * @brief Helper to clear bss section
+ */
+static void clear_bss( void ) {
+  // get bss start
+  char* clear = ( char* )&__bss_start__;
+  char* end = ( char* )&__bss_end__;
+  // clear until end
+  while ( clear < end ) {
+    *clear++ = 0;
+  }
+}
+
+/**
+ * @brief Function called before c main
+ */
+noreturn static int _main( int argc, char* argv[] ) {
+  // set environment global
+  environ = argv + argc + 1;
+  // execute main wrapped with exit
+  exit( main( argc, argv, environ ) );
+}
+
+#if __thumb__ && !defined(PREFER_THUMB)
+asm("\n"
+  ".code 32\n"
+  ".global _start\n"
+  ".type _start, %function\n"
+  "_start:\n"
+  "\tldr r0, .LC0\n"
+  "\tbx r0\n"
+  ".LC0:\n"
+  "\t.word _start_thumb\n"
+  ".size _start, .-_start\n");
+
+__attribute__((naked, used))
+static void _start_thumb(void)
+#else
+__attribute__((naked))
+void _start(void)
+#endif
+{
+  // stack register
+  register int *sp asm( "sp" );
+  // clear out bss
+  clear_bss();
+  // init / fini handling
+  #if defined( __USES_INITFINI__ )
+    // call init if existing
+    _init();
+    // register exit handler
+    atexit( _fini );
+  #endif
+  // call early main
+  _main( *sp, ( char** )( sp + 1 ) );
+}
diff --git libgloss/arm/bolthur-syscall.h libgloss/arm/bolthur-syscall.h
new file mode 100644
index 000000000..2a64c7d55
--- /dev/null
+++ libgloss/arm/bolthur-syscall.h
@@ -0,0 +1,24 @@
+
+/**
+ * Bolthur system call numbers for the ARM processor.
+ * Written by Christian Freitag <cfreitag.cf@googlemail.com>
+ * Copyright 2020 (c) Christian Freitag
+ *
+ * Permission to use, copy, modify, and distribute this software
+ * is freely granted, provided that this notice is preserved.
+ */
+
+#ifndef _LIBGLOSS_ARM_BOLTHUR_UNISTD_H
+#define _LIBGLOSS_ARM_BOLTHUR_UNISTD_H
+
+#define SYSCALL_process_exit 2
+#define SYSCALL_process_id 3
+
+#define SYSCALL_memory_acquire 21
+#define SYSCALL_memory_release 22
+
+// wrapper
+#define SYSCALL_getpid SYSCALL_process_id
+#define SYSCALL_exit SYSCALL_process_exit
+
+#endif /* _LIBGLOSS_ARM_BOLTHUR_UNISTD_H */
diff --git libgloss/arm/bolthur-syscalls0.S libgloss/arm/bolthur-syscalls0.S
new file mode 100644
index 000000000..6095182ba
--- /dev/null
+++ libgloss/arm/bolthur-syscalls0.S
@@ -0,0 +1,38 @@
+
+/**
+ * Bolthur system call numbers for the ARM processor.
+ * Written by Christian Freitag <cfreitag.cf@googlemail.com>
+ * Copyright 2020 (c) Christian Freitag
+ *
+ * Permission to use, copy, modify, and distribute this software
+ * is freely granted, provided that this notice is preserved.
+ */
+
+#include "bolthur-syscall.h"
+
+#if __thumb__
+# define FUNC(name) .type name, %function; .thumb_func; name:
+# define SET .thumb_set
+#else
+# define FUNC(name) .type name, %function; name:
+# define SET .set
+#endif
+
+#define GLOBAL(name) .global name; FUNC(name)
+#define SIZE(name) .size name, .-name
+
+# define SYSCALL(name) \
+	GLOBAL(_ ## name); \
+	svc #SYSCALL_ ## name; \
+	b _set_errno; \
+	SIZE(_ ## name)
+
+# normal system calls
+SYSCALL( memory_acquire )
+SYSCALL( memory_release )
+SYSCALL( process_id )
+SYSCALL( process_exit )
+
+# wrapped system calls
+SYSCALL( getpid )
+SYSCALL( exit )
diff --git libgloss/arm/bolthur-syscalls1.c libgloss/arm/bolthur-syscalls1.c
new file mode 100644
index 000000000..245803d4e
--- /dev/null
+++ libgloss/arm/bolthur-syscalls1.c
@@ -0,0 +1,30 @@
+
+/**
+ * Bolthur system call numbers for the ARM processor.
+ * Written by Christian Freitag <cfreitag.cf@googlemail.com>
+ * Copyright 2020 (c) Christian Freitag
+ *
+ * Permission to use, copy, modify, and distribute this software
+ * is freely granted, provided that this notice is preserved.
+ */
+
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/fcntl.h>
+#include <sys/times.h>
+#include <sys/errno.h>
+#include <sys/time.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <errno.h>
+
+#undef errno
+extern int errno;
+
+int _set_errno( int n ) {
+  if ( n < 0 ) {
+    errno = -n;
+    return -1;
+  }
+  return n;
+}
diff --git libgloss/arm/elf-bolthur.specs libgloss/arm/elf-bolthur.specs
new file mode 100644
index 000000000..d2ad3e850
--- /dev/null
+++ libgloss/arm/elf-bolthur.specs
@@ -0,0 +1,11 @@
+%rename lib libc
+
+*libgloss:
+-lgloss-bolthur
+
+*lib:
+%(libc) %(libgloss)
+
+*startfile:
+bolthur-crt0%O%s crti%O%s crtbegin%O%s
+
diff --git newlib/configure.host newlib/configure.host
index a84c0c80a..b5cce828e 100644
--- newlib/configure.host
+++ newlib/configure.host
@@ -191,11 +191,11 @@ case "${host_cpu}" in
   i[34567]86)
 	# Don't use for these since they provide their own setjmp.
 	case ${host} in
-	*-*-sco* | *-*-cygwin*) 
+	*-*-sco* | *-*-cygwin*)
 		libm_machine_dir=i386
 		machine_dir=i386
 		;;
-	*) 
+	*)
 		libm_machine_dir=i386
 		machine_dir=i386
 		mach_add_setjmp=true
@@ -211,7 +211,7 @@ case "${host_cpu}" in
 	machine_dir=lm32
 	;;
   m32c)
-	machine_dir=m32c	
+	machine_dir=m32c
 	newlib_cflags="${newlib_cflags} -DPREFER_SIZE_OVER_SPEED -DSMALL_MEMORY"
 	;;
 
@@ -224,7 +224,7 @@ case "${host_cpu}" in
 	newlib_cflags="-DPREFER_SIZE_OVER_SPEED -Os -mrelax"
 	CFLAGS="-g -Os"
 	;;
-  
+
   m68*)
 	machine_dir=m68k
 	newlib_cflags="${newlib_cflags} -DCOMPACT_CTYPE"
@@ -429,6 +429,11 @@ esac
 # THIS TABLE IS ALPHA SORTED.  KEEP IT THAT WAY.
 
 case "${host}" in
+  *-*-bolthur*)
+  sys_dir=bolthur
+  # crt0 is provided by libgloss.
+  have_crt0="no"
+  ;;
   *-*-cygwin*)
 	posix_dir=posix
 	stdio64_dir=stdio64
@@ -528,7 +533,7 @@ case "${host}" in
 	unix_dir=unix
 	posix_dir=posix
 	crt1=crt1.o
-	crt1_dir=libc/sys/${sys_dir}	
+	crt1_dir=libc/sys/${sys_dir}
 	gcc_dir=`gcc -print-search-dirs | awk '/^install:/{print $2}'`
 	default_newlib_io_c99_formats="yes"
 	default_newlib_io_long_double="yes"
@@ -633,6 +638,9 @@ esac
 # THIS TABLE IS ALPHA SORTED.  KEEP IT THAT WAY.
 
 case "${host}" in
+	*-*-bolthur*)
+	#newlib_cflags="${newlib_cflags} -DSIGNAL_PROVIDED"
+	;;
   *-*-cygwin*)
 	test -z "$cygwin_srcdir" && cygwin_srcdir=`cd ${srcdir}/../winsup/cygwin; pwd`
 	export cygwin_srcdir
@@ -662,7 +670,7 @@ case "${host}" in
 	newlib_cflags="${newlib_cflags} -ffunction-sections -fdata-sections "
 	newlib_cflags="${newlib_cflags} -D_COMPILING_NEWLIB"
 newlib_cflags="${newlib_cflags} -DCLOCK_PROVIDED -DMALLOC_PROVIDED -DEXIT_PROVIDED -DSIGNAL_PROVIDED -DGETREENT_PROVIDED -DREENTRANT_SYSCALLS_PROVIDED -DHAVE_NANOSLEEP -DHAVE_BLKSIZE -DHAVE_FCNTL -DHAVE_ASSERT_FUNC"
-        # turn off unsupported items in posix directory 
+        # turn off unsupported items in posix directory
 	newlib_cflags="${newlib_cflags} -D_NO_GETLOGIN -D_NO_GETPWENT -D_NO_GETUT -D_NO_GETPASS -D_NO_SIGSET -D_NO_WORDEXP -D_NO_POPEN -D_NO_POSIX_SPAWN"
 	;;
 # VxWorks supplies its own version of malloc, and the newlib one
@@ -749,17 +757,17 @@ newlib_cflags="${newlib_cflags} -DCLOCK_PROVIDED -DMALLOC_PROVIDED -DEXIT_PROVID
         syscall_dir=syscalls
         ;;
   h8300*-*-*)
-	syscall_dir=syscalls	
+	syscall_dir=syscalls
 	default_newlib_io_long_long="yes"
 	newlib_cflags="${newlib_cflags} -DSMALL_DTOA -DSMALL_MEMORY"
 	# Simulator only extensions for H8300.
 	# Uncomment the next line to enable them.
 	# newlib_cflags="${newlib_cflags} -D__SIMULATOR__"
-	;;	
+	;;
   h8500-*-*)
-	syscall_dir=syscalls	
+	syscall_dir=syscalls
 	newlib_cflags="${newlib_cflags} -DSMALL_DTOA -DSMALL_MEMORY"
-	;;	
+	;;
   i[34567]86-*-sco*)
 	newlib_cflags="${newlib_cflags} -DSIGNAL_PROVIDED -DHAVE_FCNTL"
 	;;
@@ -906,7 +914,7 @@ newlib_cflags="${newlib_cflags} -DCLOCK_PROVIDED -DMALLOC_PROVIDED -DEXIT_PROVID
 	syscall_dir=
 	;;
   w65-*-*)
-	syscall_dir=syscalls	
+	syscall_dir=syscalls
 	newlib_cflags="${newlib_cflags} -DSMALL_DTOA -DSMALL_MEMORY"
 	;;
   xc16x-*)
@@ -957,7 +965,7 @@ fi
 # Disable atexit dynamic allocation if requested.
 if [ "x${newlib_atexit_dynamic_alloc}" = "x" ]; then
 	if [ ${default_newlib_atexit_dynamic_alloc} = "yes" ]; then
-		newlib_atexit_dynamic_alloc="yes"; 
+		newlib_atexit_dynamic_alloc="yes";
 	fi
 fi
 
diff --git newlib/libc/sys/bolthur/Makefile.am newlib/libc/sys/bolthur/Makefile.am
new file mode 100644
index 000000000..e97f5f24f
--- /dev/null
+++ newlib/libc/sys/bolthur/Makefile.am
@@ -0,0 +1,15 @@
+
+AUTOMAKE_OPTIONS = cygnus
+
+INCLUDES = $(NEWLIB_CFLAGS) $(CROSS_CFLAGS) $(TARGET_CFLAGS)
+
+AM_CCASFLAGS = $(INCLUDES)
+
+noinst_LIBRARIES = lib.a
+
+lib_a_SOURCES = highlevel.c
+lib_a_CCASFLAGS = $(AM_CCASFLAGS)
+lib_a_CFLAGS = $(AM_CFLAGS)
+
+ACLOCAL_AMFLAGS = -I ../../.. -I ../../../..
+CONFIG_STATUS_DEPENDENCIES = $(newlib_basedir)/configure.host
diff --git newlib/libc/sys/bolthur/configure.in newlib/libc/sys/bolthur/configure.in
new file mode 100644
index 000000000..712ed2b9f
--- /dev/null
+++ newlib/libc/sys/bolthur/configure.in
@@ -0,0 +1,12 @@
+
+AC_PREREQ(2.59)
+AC_INIT([newlib],[NEWLIB_VERSION])
+AC_CONFIG_SRCDIR([highlevel.c])
+
+dnl Can't be done in NEWLIB_CONFIGURE because that confuses automake.
+AC_CONFIG_AUX_DIR(../../../..)
+
+NEWLIB_CONFIGURE(../../..)
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
diff --git newlib/libc/sys/bolthur/highlevel.c newlib/libc/sys/bolthur/highlevel.c
new file mode 100644
index 000000000..05cebff83
--- /dev/null
+++ newlib/libc/sys/bolthur/highlevel.c
@@ -0,0 +1,123 @@
+
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/fcntl.h>
+#include <sys/times.h>
+#include <sys/errno.h>
+#include <sys/time.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <errno.h>
+
+#define MAX_POOL_ENTRIES 100
+
+static int next_pool;
+void* pool[ MAX_POOL_ENTRIES ];
+
+extern caddr_t _memory_acquire( int, int );
+extern void _memory_release( uintptr_t, int );
+
+/**
+ * @brief sbrk fake syscall ( non contiguous )
+ *
+ * @param incr
+ * @return caddr_t
+ */
+caddr_t _sbrk( int incr ) {
+  void *ptr = 0;
+  static caddr_t sbrk_top = 0;
+
+  if ( incr > 0 ) {
+    // allocate memory
+    ptr = _memory_acquire( incr, 2 );
+    // handle error
+    if ( 0 == ptr ) {
+      return ( caddr_t )-1;
+    }
+    // save for free during cleanup
+    pool[ next_pool++ ] = ptr;
+    // FIXME: ADJUST HERE
+    //ptr = ( caddr_t )( ( ( ( size_t )ptr ) + RM_PAGE_MASK ) & ~RM_PAGE_MASK );
+    // set new top
+    sbrk_top = ( caddr_t )ptr + incr;
+    // return pointer
+    return ptr;
+  // shrink is not supported
+  } else if ( incr < 0 ) {
+    return ( void* )-1;
+  } else {
+    return sbrk_top;
+  }
+}
+
+int _close( int file ) {
+  return -1;
+}
+
+int _execve( char* name, char** argv, char** env ) {
+  errno = ENOMEM;
+  return -1;
+}
+
+int _fork() {
+  errno = EAGAIN;
+  return -1;
+}
+
+int _fstat( int file, struct stat* st ) {
+  st->st_mode = S_IFCHR;
+  return 0;
+}
+
+int _isatty( int file ) {
+  return 1;
+}
+
+int _kill( int pid, int sig ) {
+  errno = EINVAL;
+  return -1;
+}
+
+int _link( char* old, char* new ) {
+  errno = EMLINK;
+  return -1;
+}
+
+int _lseek( int file, int ptr, int dir ) {
+  return 0;
+}
+
+int _open( const char* name, int flags, ... ) {
+  return -1;
+}
+
+int _read( int file, char* ptr, int len ) {
+  return 0;
+}
+
+int _stat( const char* file, struct stat* st ) {
+  st->st_mode = S_IFCHR;
+  return 0;
+}
+
+clock_t _times( struct tms* buf ) {
+  return -1;
+}
+
+int _unlink( char* name ) {
+  errno = ENOENT;
+  return -1;
+}
+
+int _wait( int* status ) {
+  errno = ECHILD;
+  return -1;
+}
+
+int _write( int file, char* ptr, int len ) {
+  return -1;
+}
+
+int _gettimeofday( struct timeval* p, struct timezone* z ) {
+  return -1;
+}
diff --git newlib/libc/sys/configure.in newlib/libc/sys/configure.in
index a65d1e71f..44a6528df 100644
--- newlib/libc/sys/configure.in
+++ newlib/libc/sys/configure.in
@@ -5,7 +5,7 @@ AC_PREREQ(2.59)
 AC_INIT([newlib],[NEWLIB_VERSION])
 AC_CONFIG_SRCDIR([a29khif])
 
-dnl Can't be done in NEWLIB_CONFIGURE because that confuses automake. 
+dnl Can't be done in NEWLIB_CONFIGURE because that confuses automake.
 AC_CONFIG_AUX_DIR(../../..)
 
 NEWLIB_CONFIGURE(../..)
@@ -50,6 +50,7 @@ if test -n "${sys_dir}"; then
 	tirtos) AC_CONFIG_SUBDIRS(tirtos) ;;
 	w65) AC_CONFIG_SUBDIRS(w65) ;;
 	z8ksim) AC_CONFIG_SUBDIRS(z8ksim) ;;
+	bolthur) AC_CONFIG_SUBDIRS(bolthur) ;;
   esac;
 fi
 
