diff --git m4/libtool.m4 m4/libtool.m4
index 96e6c7d0..3a4c944f 100644
--- m4/libtool.m4
+++ m4/libtool.m4
@@ -3131,6 +3131,18 @@ netbsd*)
   stdlibflag=
   ;;
 
+bolthur*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  dynamic_linker='Bolthur ld-bolthur.so'
+  shlibpath_var=LD_LIBRARY_PATH
+  shlibpath_overrides_runpath=no
+  hardcode_into_libs=yes
+  ;;
+
 newsos6)
   version_type=linux # correct to gnu/linux during the next big refactor
   library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
@@ -3764,6 +3776,10 @@ haiku*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+bolthur*)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 hpux10.20* | hpux11*)
   lt_cv_file_magic_cmd=$FILECMD
   case $host_cpu in
