diff --git lib/handle.c lib/handle.c
index fce84be..f07be3d 100644
--- lib/handle.c
+++ lib/handle.c
@@ -49,7 +49,7 @@ tar_init(TAR **t, const char *pathname, tartype_t *type,
 	if (*t == NULL)
 		return -1;
 
-	(*t)->pathname = pathname;
+	(*t)->pathname = ( char* )pathname;
 	(*t)->options = options;
 	(*t)->type = (type ? type : &default_type);
 	(*t)->oflags = oflags;
@@ -133,5 +133,3 @@ tar_close(TAR *t)
 
 	return i;
 }
-
-
diff --git libtar/Makefile.in libtar/Makefile.in
index 7c6f1bc..2b2c748 100644
--- libtar/Makefile.in
+++ libtar/Makefile.in
@@ -20,7 +20,7 @@ SHELL           = @SHELL@
 
 ### Installation programs and flags
 INSTALL		= @INSTALL@
-INSTALL_PROGRAM	= @INSTALL_PROGRAM@ -s
+INSTALL_PROGRAM	= @INSTALL_PROGRAM@
 INSTALL_DATA	= @INSTALL_DATA@
 LN_S		= @LN_S@
 MKDIR		= @MKDIR@
@@ -75,4 +75,3 @@ distclean: clean
 install: ${ALL}
 	${MKDIR} ${DESTDIR}${bindir}
 	$(LIBTOOL) --mode=install ${INSTALL_PROGRAM} libtar ${DESTDIR}${bindir}
-
