diff --git .gitignore .gitignore
index 07d4a12..1d2375b 100644
--- .gitignore
+++ .gitignore
@@ -1,3 +1,5 @@
+build/
+
 *~
 *.gz
 *.Z
diff --git abi-tags abi-tags
index 87cd681..802e0e7 100644
--- abi-tags
+++ abi-tags
@@ -18,6 +18,8 @@
 
 .*-.*-linux.*		0	2.0.0	# earliest compatible kernel version
 
+.*-.*-bolthur.*		1	0.0.0
+
 .*-.*-gnu-gnu.*		1	0.0.0
 
 .*-sun-solaris2.*	2	2.0.0	# just an arbitrary value
diff --git configure configure
index 4795e72..a1821cd 100755
--- configure
+++ configure
@@ -3726,7 +3726,7 @@ fi
 # $host_os of `gnu*' here to be `gnu-gnu*' just so that it can tell.
 # This doesn't get used much beyond that, so it's fairly safe.
 case "$host_os" in
-linux*)
+linux* | bolthur*)
   ;;
 gnu*)
   host_os=`echo $host_os | sed -e 's/gnu/gnu-gnu/'`
@@ -3833,7 +3833,7 @@ fi
 ###
 if test -z "$enable_hacker_mode" && test x"$libc_config_ok" != xyes; then
   case "$machine-$host_os" in
-  *-linux* | *-gnu*)
+  *-linux* | *-bolthur* | *-gnu*)
     ;;
   *)
     as_fn_error $? "
@@ -4153,7 +4153,7 @@ os="`echo $os | sed 's/\([0-9A-Z]\)[v_]\([0-9A-Z]\)/\1.\2/g'`"
 test "x$base_os" != x || case "$os" in
 gnu*)
   base_os=mach/hurd ;;
-linux*)
+linux* | bolthur*)
   base_os=unix/sysv ;;
 esac
 
@@ -4712,7 +4712,7 @@ fi
 
 # These programs are version sensitive.
 
-for ac_prog in gnumake gmake make
+for ac_prog in gmake gnumake make
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog; ac_word=$2
diff --git configure.ac configure.ac
index 93e68fb..b852ef5 100644
--- configure.ac
+++ configure.ac
@@ -438,7 +438,7 @@ fi
 # $host_os of `gnu*' here to be `gnu-gnu*' just so that it can tell.
 # This doesn't get used much beyond that, so it's fairly safe.
 case "$host_os" in
-linux*)
+linux* | bolthur*)
   ;;
 gnu*)
   host_os=`echo $host_os | sed -e 's/gnu/gnu-gnu/'`
@@ -515,7 +515,7 @@ LIBC_PRECONFIGURE([$srcdir], [for sysdeps])
 ###
 if test -z "$enable_hacker_mode" && test x"$libc_config_ok" != xyes; then
   case "$machine-$host_os" in
-  *-linux* | *-gnu*)
+  *-linux* | *-bolthur* | *-gnu*)
     ;;
   *)
     AC_MSG_ERROR([
@@ -752,7 +752,7 @@ os="`echo $os | sed 's/\([0-9A-Z]\)[v_]\([0-9A-Z]\)/\1.\2/g'`"
 test "x$base_os" != x || case "$os" in
 gnu*)
   base_os=mach/hurd ;;
-linux*)
+linux* | bolthur*)
   base_os=unix/sysv ;;
 esac
 
@@ -979,7 +979,7 @@ fi
 
 # These programs are version sensitive.
 AC_CHECK_TOOL_PREFIX
-AC_CHECK_PROG_VER(MAKE, gnumake gmake make, --version,
+AC_CHECK_PROG_VER(MAKE, gmake gnumake make, --version,
   [GNU Make[^0-9]*\([0-9][0-9.]*\)],
   [[4-9].* | [1-9][0-9]*], critic_missing="$critic_missing make")
 
